### PR TITLE
Custom kind removal

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Removes the `kind` kwarg from custom functions, addressing issue #635 and partially addresses #528.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality
      to not work as expected)
- This change requires a documentation update

# How Has This Been Tested?

Created new and adapted existing unit tests.  Check out `test_custom.py` for examples.

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

## Downstream modules
- [x] pysatKamodo @aburrell
- [x] pysatMissions @jklenzing 
- [x] pysatModels @aburrell - https://github.com/pysat/pysatModels/pull/70
- [x] pysatSeasons @rstoneback 
- [x] pysatCDAAC @jklenzing 
- [x] pysatMadrigal @aburrell 
- [x] pysatNASA @jklenzing 
- [x] pysatSpaceWeather @aburrell - https://github.com/pysat/pysatSpaceWeather/pull/40
- [x] pysatIncubator @rstoneback - https://github.com/pysat/pysatIncubator/pull/7

### CODE REVIEW

**Comment by jklenzing** [comment]
`custom_attach` not used in pysatNASA or pysatCDAAC.  All instances in pysatMissions do not use the `kind` kwarg.

File: pysat/_instrument.py [hunk 1]
```diff
@@ -1962,168 +1915,54 @@ def custom_attach(self, function, kind='modify', at_pos='end', args=[],
...
-                    elif kind == 'pass':
-                        tempd = self.copy()
-                        null_out = func(tempd, *arg, **kwarg)
-                        del tempd
-                        if null_out is not None:
-                            raise ValueError(''.join(('Pass functions should ',
-                                                      'not return any ',
-                                                      'information via ',
-                                                      'return.')))
+                    # Custom functions do nothing or modify loaded data. Methods
```
Comment (rstoneback) [inline pysat/_instrument.py:1945]: My first thought for this issue was we would put a DeprecationWarning for any values of 'kind' that aren't 'modify', but leave the functionality in until the next major release. 

Taking the code out reduces our code weight which *is* nice, but given the other deprecations I'm surprised.

**Review by rstoneback — COMMENTED** [review]
Thanks @aburrell!

File: pysat/_instrument.py (hunk 1 above)
Comment (aburrell) [reply pysat/_instrument.py:1945]: I don't see any point in making the huge change of moving Custom to be a part of Instrument here and delaying the small change of removing the `kind` kwarg.  There will obviously be a deprecation warning in the last 2.X release, though.

**Review by rstoneback — APPROVED** [review]

**Pull request merged**